0xPolygon/polygon-edge
# Index methods by signature when there is ambiguity

# Description

This PR is a continuation of #1854 and it generates SC bindings in a way that they index ABI methods by signature in case the function was resolved by signature, instead of by its name (e.g. `ChildERC20PredicateAccessList` defines two functions with the same name but different params, which caused us some troubles in the past).

# Changes include


# Checklist


## Testing


### Manual tests

1. In the `ChildERC20PredicateAccessList.json` artifact within the core-contracts, make sure that initialize function with 5 parameters parameters is in the first place (before the one with 8 of them, that is initializing access-list configuration)
2. Generate artifacts and bindings in the Edge, by running `make generate-smart-contract-bindings`
3. Run `TestE2E_Bridge_Transfers_AccessLists` and make sure it is executed successfully (namely it does not fail after first withdrawal being made)
